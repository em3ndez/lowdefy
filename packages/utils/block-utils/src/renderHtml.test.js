/*
  Copyright 2020-2024 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import { render } from '@testing-library/react';

import renderHtml from './renderHtml.js';
import makeCssClass from './makeCssClass.js';
const methods = {
  makeCssClass,
};

test('renderHtml html is undefined', () => {
  expect(renderHtml({ methods })).toBe(undefined);
  expect(renderHtml({ html: undefined, methods })).toBe(undefined);
});

test('renderHtml html string', () => {
  const { container } = render(renderHtml({ html: '<p>Hello</p>', methods }));
  expect(container.firstChild).toMatchSnapshot();
});

test('renderHtml html number 0', () => {
  const { container } = render(renderHtml({ html: 0, methods }));
  expect(container.firstChild).toMatchSnapshot();
});

test('renderHtml html number 123', () => {
  const { container } = render(renderHtml({ html: 123, methods }));
  expect(container.firstChild).toMatchSnapshot();
});

test('renderHtml html boolean false', () => {
  const { container } = render(renderHtml({ html: false, methods }));
  expect(container.firstChild).toMatchSnapshot();
});

test('renderHtml html boolean true', () => {
  const { container } = render(renderHtml({ html: true, methods }));
  expect(container.firstChild).toMatchSnapshot();
});
