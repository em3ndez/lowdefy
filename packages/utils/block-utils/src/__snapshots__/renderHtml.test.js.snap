// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderHtml html boolean false 1`] = `
<span
  class="emotion-0"
>
  false
</span>
`;

exports[`renderHtml html boolean true 1`] = `
<span
  class="emotion-0"
>
  true
</span>
`;

exports[`renderHtml html number 0 1`] = `
<span
  class="emotion-0"
>
  0
</span>
`;

exports[`renderHtml html number 123 1`] = `
<span
  class="emotion-0"
>
  123
</span>
`;

exports[`renderHtml html string 1`] = `
<span
  class="emotion-0"
>
  <p>
    Hello
  </p>
</span>
`;
